### PR TITLE
Rework Movie Metadata data model

### LINKED ISSUE #5489: {Movie.CleanOriginalTitle}
Requesting for this filename token to be added. Right now we have CleanTitle and OriginalTitle, and I don't see why we have to choose between the two and not also have the option of both functionalities at the same time. 

Example:
- Using {Movie.CleanTitle}
The Norwegian film Den 12. mann is renamed to the non-native title of `The.12th.Man`, which is undesired.
Mission: Impossible - Fallout is renamed to `Mission.Impossible.Fallout` which is good.

- Using {Movie.OriginalTitle}
Den 12. mann is renamed to `Den.12.mann` which is good.
Mission: Impossible - Fallout is renamed to `Mission.-.Impossible.-.Fallout` which is undesired.

CleanOriginalTitle would then provide the best of both worlds.

Thanks for considering.

[AB#1023](https://dev.azure.com/Servarr/7ab38f4e-5a57-4d70-84f4-94dd9bc5d6df/_workitems/edit/1023)

### PR DESCRIPTION
**Pull request by Qstick**
#### Database Migration
YES - 207

#### Description
Move metadata off Movie table and onto MovieMetadata table. This avoids us from storing metadata in 3 locations in DB (Collections, Movie, ImportListMovie)

#### Todos
- [x] DB Migration Tests
- [x] Alot of testing to make sure nothing is broken
- [x] Migration and rework of ImportListMovies Table
- [ ] Shared MovieMeta Refresh
- [x] Speed Tests

Fixes #5489
Data for #1947

### CODE REVIEW

File: src/NzbDrone.Core/Datastore/Migration/207_movie_metadata.cs [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+                          SELECT TmdbId, ImdbId, Title, SortTitle, CleanTitle, OriginalTitle, CleanTitle, OriginalLanguage, Overview, Status, LastInfoSync, Images, Genres, Ratings, Runtime, InCinemas, PhysicalRelease, DigitalRelease, Year, SecondaryYear, Recommendations, Certification, YouTubeTrailerId, Studio, Collection, Website
+                          FROM Movies");
+
+            Execute.Sql(@"INSERT INTO MovieMetadata (TmdbId, ImdbId, Title, SortTitle, CleanTitle, OriginalTitle, CleanOriginalTitle, OriginalLanguage, Overview, Status, LastInfoSync, Images, Genres, Ratings, Runtime, InCinemas, PhysicalRelease, DigitalRelease, Year, Certification, YouTubeTrailerId, Studio, Collection, Website)
+                          SELECT TmdbId, ImdbId, Title, SortTitle, Title, OriginalTitle, OriginalTitle, 1, Overview, Status, LastInfoSync, Images, Genres, Ratings, Runtime, InCinemas, PhysicalRelease, DigitalRelease, Year, Certification, YouTubeTrailerId, Studio, Collection, Website
+                          FROM ImportListMovies
+                          WHERE ImportListMovies.TmdbId NOT IN ( SELECT MovieMetadata.TmdbId FROM MovieMetadata )");
```
Comment (RobinDadswell) [inline src/NzbDrone.Core/Datastore/Migration/207_movie_metadata.cs:46]: Postgres support on these please dude :D (and the various other SQL stuff in this migration)

File: frontend/src/Settings/MediaManagement/Naming/NamingModal.js [hunk 2]
```diff
@@ -116,10 +116,11 @@ class NamingModal extends Component {
 
     const movieTokens = [
       { token: '{Movie Title}', example: 'Movie\'s Title' },
-      { token: '{Movie Title:DE}', example: 'Filetitle' },
+      { token: '{Movie Title:DE}', example: 'Titel des Films' },
       { token: '{Movie CleanTitle}', example: 'Movies Title' },
       { token: '{Movie TitleThe}', example: 'Movie\'s Title, The' },
       { token: '{Movie OriginalTitle}', example: 'Τίτλος ταινίας' },
```
Comment (bakerboy448) [inline frontend/src/Settings/MediaManagement/Naming/NamingModal.js:122]: need a special character here to demonstrate what CleanOriginalTitlec does

File: src/NzbDrone.Core/Datastore/Migration/207_movie_metadata.cs [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+                .WithColumn("Status").AsInt32()
+                .WithColumn("LastInfoSync").AsDateTime().Nullable()
+                .WithColumn("Runtime").AsInt32()
+                .WithColumn("InCinemas").AsDateTime().Nullable()
+                .WithColumn("PhysicalRelease").AsDateTime().Nullable()
+                .WithColumn("DigitalRelease").AsDateTime().Nullable()
+                .WithColumn("Year").AsInt32().Nullable()
+                .WithColumn("SecondaryYear").AsInt32().Nullable()
+                .WithColumn("Ratings").AsString().Nullable()
+                .WithColumn("Recommendations").AsString().Nullable()
```
Comment (Qstick) [inline src/NzbDrone.Core/Datastore/Migration/207_movie_metadata.cs:31]: Not Nullable

File: src/NzbDrone.Core/Datastore/Migration/207_movie_metadata.cs [hunk 4]
```diff
@@ -0,0 +1,150 @@
...
+            Execute.Sql(@"INSERT INTO ""MovieMetadata"" (""TmdbId"", ""ImdbId"", ""Title"", ""SortTitle"", ""CleanTitle"", ""OriginalTitle"", ""CleanOriginalTitle"", ""OriginalLanguage"", ""Overview"", ""Status"", ""LastInfoSync"", ""Images"", ""Genres"", ""Ratings"", ""Runtime"", ""InCinemas"", ""PhysicalRelease"", ""DigitalRelease"", ""Year"", ""Certification"", ""YouTubeTrailerId"", ""Studio"", ""Collection"", ""Website"")
+                          SELECT ""TmdbId"", ""ImdbId"", ""Title"", ""SortTitle"", ""Title"", ""OriginalTitle"", ""OriginalTitle"", 1, ""Overview"", ""Status"", ""LastInfoSync"", ""Images"", ""Genres"", ""Ratings"", ""Runtime"", ""InCinemas"", ""PhysicalRelease"", ""DigitalRelease"", ""Year"", ""Certification"", ""YouTubeTrailerId"", ""Studio"", ""Collection"", ""Website""
+                          FROM ""ImportListMovies""
+                          WHERE ""ImportListMovies"".""TmdbId"" NOT IN ( SELECT ""MovieMetadata"".""TmdbId"" FROM ""MovieMetadata"" )");
+
+            // Add an MovieMetadataId column to Movies
+            Alter.Table("Movies").AddColumn("MovieMetadataId").AsInt32().WithDefaultValue(0);
```
Comment (Qstick) [inline src/NzbDrone.Core/Datastore/Migration/207_movie_metadata.cs:52]: Unique

**Comment by bakerboy448** [comment]
Wiki - https://github.com/Servarr/Wiki/commit/edaf0b474a501e362b0e371dc8bd44cbf6840e98

**Pull request merged**